### PR TITLE
Update schemas to include new indexes

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

New indexes were missing from the previously generated schemas in https://github.com/mattermost/mattermost-server/pull/18223

Not sure how the merge didn't cause a failure though. Maybe we are not running the schema check step over there.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by metanerd** [comment]
Sorry, I don't quite understand. What do you mean by over there? Do you mean this? https://github.com/mattermost/mattermost-server/blob/ac252f0771de25d056220b562297a04e1e40eeb0/.gitlab-ci.yml#L128

**Comment by streamer45** [comment]
> Sorry, I don't quite understand. What do you mean by over there? Do you mean this?
> 
> https://github.com/mattermost/mattermost-server/blob/ac252f0771de25d056220b562297a04e1e40eeb0/.gitlab-ci.yml#L128

I was expecting both the original PR (https://github.com/mattermost/mattermost-server/pull/18223) and the subsequent merge into master (https://github.com/mattermost/mattermost-server/commit/ac252f0771de25d056220b562297a04e1e40eeb0) to trigger a failure given the schema is missing things.

**Comment by metanerd** [comment]
OK. The schema tests ran in both CircleCI and GitLab and it succeeded in both with "Both schemas are same", not with some failure and pipeline still succeeds. I don't know enough to check what's causing it.

**Comment by streamer45** [comment]
> OK. The schema tests ran in both CircleCI and GitLab and it succeeded in both with "Both schemas are same", not with some failure and pipeline still succeeds. I don't know enough to check what's causing it.

Need to verify, maybe there's some more difference in the checks between CI and local. The `make test-db-migration` is currently failing on master if you try locally.

File: scripts/mattermost-mysql-6.0.sql [hunk 1]
```diff
@@ -1,15 +1,19 @@
--- MySQL dump 10.13  Distrib 5.7.35, for Linux (x86_64)
+-- MariaDB dump 10.19  Distrib 10.5.10-MariaDB, for Linux (x86_64)
```
Comment (agnivade) [inline scripts/mattermost-mysql-6.0.sql:1]: Hmm .. why MariaDB though? Does it match with CI?

**Review by agnivade — APPROVED** [review]
Looks good, but we should look into why CI acts differently.

File: scripts/mattermost-mysql-6.0.sql (hunk 1 above)
Comment (streamer45) [reply scripts/mattermost-mysql-6.0.sql:1]: That's just my local client, it's a comment though.

**Comment by streamer45** [comment]
> Looks good, but we should look into why CI acts differently.

Yes, me and @metanerd are having a look.

File: scripts/mattermost-mysql-6.0.sql (hunk 1 above)
Comment (agnivade) [reply scripts/mattermost-mysql-6.0.sql:1]: Ah nvm, I thought we were matching the final output with this text file. Now I see that we just use this file for importing and then dump it again.

Comment (streamer45) [reply scripts/mattermost-mysql-6.0.sql:1]: Yeah, it's just an input "template" to initialize the migrated db.

**Comment by streamer45** [comment]
By the way, something I've found out while working on this that may be of note is that the schema checking script doesn't really test migrations cause they never get to run. In fact, the schema version (the one saved in the Systems table) is missing on the newly created database so we set it to whatever the server version is and no upgrade executes due to: https://github.com/mattermost/mattermost-server/blob/b6564fec6d62c4f9b3a742748cd2c2364f049646/store/sqlstore/upgrade.go#L126-L135

I am going to assume this is the intended behaviour since it's been  like this for a very long time but I'd also consider adding an additional step that runs the migrations and compares the final output. Of course this may be all unnecessary coming the new system.

**Review by metanerd — APPROVED** [review]
Sorry, forgot to approve!

**Pull request merged**